node/src/keyshare/mod.rs

```diff
@@ -1,80 +1,396 @@
-pub mod gcp;
+pub mod compat;
+mod gcp;
 pub mod local;
+pub mod permanent;
+mod temporary;
+#[cfg(test)]
+pub mod test_utils;
 
-use cait_sith::KeygenOutput;
-use k256::{AffinePoint, Scalar, Secp256k1};
+use crate::hkdf::affine_point_to_public_key;
+use anyhow::Context;
+use k256::{AffinePoint, Scalar};
+use mpc_contract::primitives::key_state::Keyset;
+use mpc_contract::primitives::key_state::{EpochId, KeyEventId, KeyForDomain};
+use permanent::{PermanentKeyStorage, PermanentKeyStorageBackend, PermanentKeyshareData};
 use serde::{Deserialize, Serialize};
+use temporary::TemporaryKeyStorage;
 
-/// The root keyshare data along with an epoch. The epoch is incremented
-/// for each key resharing. This is the format stored in the old MPC
-/// implementation, and we're keeping it the same to ease migration.
-#[derive(Clone, Serialize, Deserialize)]
-pub struct RootKeyshareData {
-    pub epoch: u64,
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Secp256k1KeyshareData {
     pub private_share: Scalar,
     pub public_key: AffinePoint,
 }
 
-impl RootKeyshareData {
-    pub fn keygen_output(&self) -> KeygenOutput<Secp256k1> {
-        KeygenOutput {
-            private_share: self.private_share,
-            public_key: self.public_key,
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub enum KeyshareData {
+    Secp256k1(Secp256k1KeyshareData),
```

Comment:
Wouldn't it be better to link it to the existing type of a `SignatureProvider::KeygenOutput`? To avoid duplications and further type-conversions